File: clients/java/src/main/java/com/sentinel/hunter/SentinelClient.java
package com.sentinel.hunter;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

/**
 * Enterprise Java SDK Client for Sentinel-AutoGen-Hunter Threat Orchestration Gateway.
 */
public class SentinelClient {

    private final HttpClient httpClient;
    private final String baseUrl;

    public SentinelClient(String baseUrl) {
        this.baseUrl = baseUrl.endsWith("/") ? baseUrl.substring(0, baseUrl.length() - 1) : baseUrl;
        this.httpClient = HttpClient.newBuilder()
                .connectTimeout(Duration.ofSeconds(10))
                .build();
    }

    public SentinelClient() {
        this("http://localhost:8000");
    }

    /**
     * Checks orchestrator health.
     */
    public String checkHealth() throws IOException, InterruptedException {
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(baseUrl + "/health"))
                .GET()
                .build();
        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        return response.body();
    }

    /**
     * Ingests telemetry event for autonomous multi-agent analysis.
     */
    public String ingestTelemetry(String service, String ip, String rawLog) throws IOException, InterruptedException {
        String jsonPayload = String.format(
                "{\"service\":\"%s\",\"event_type\":\"auth_attempt\",\"source_ip\":\"%s\",\"raw_log\":\"%s\"}",
                service, ip, rawLog.replace("\"", "\\\"")
        );

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(baseUrl + "/api/v1/telemetry"))
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString(jsonPayload))
                .build();

        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
        return response.body();
    }
}

File: clients/dotnet/SentinelClient.cs
using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace Sentinel.AutoGen.Hunter
{
    /// <summary>
    /// Enterprise .NET SDK Client for Sentinel-AutoGen-Hunter Threat Orchestration Gateway.
    /// </summary>
    public class SentinelClient
    {
        private readonly HttpClient _httpClient;
        private readonly string _endpoint;

        public SentinelClient(string endpoint = "http://localhost:8000", HttpClient? httpClient = null)
        {
            _endpoint = endpoint.TrimEnd('/');
            _httpClient = httpClient ?? new HttpClient();
        }

        /// <summary>
        /// Retrieves health status from the orchestrator engine.
        /// </summary>
        public async Task<string> GetHealthAsync()
        {
            var response = await _httpClient.GetAsync($"{_endpoint}/health");
            response.EnsureSuccessStatusCode();
            return await response.Content.ReadAsStringAsync();
        }

        /// <summary>
        /// Sends telemetry for autonomous multi-agent analysis.
        /// </summary>
        public async Task<string> ScanTelemetryAsync(string service, string ip, string rawLog)
        {
            var json = $"{{\"service\":\"{service}\",\"event_type\":\"auth_attempt\",\"source_ip\":\"{ip}\",\"raw_log\":\"{rawLog}\"}}";
            var content = new StringContent(json, Encoding.UTF8, "application/json");
            var response = await _httpClient.PostAsync($"{_endpoint}/api/v1/telemetry", content);
            response.EnsureSuccessStatusCode();
            return await response.Content.ReadAsStringAsync();
        }

        /// <summary>
        /// Interacts with the Sentinel Gateway OpenClaw-compatible Chat Completions API.
        /// </summary>
        public async Task<string> PromptSwarmAsync(string prompt)
        {
            var json = $"{{\"model\":\"sentinel-autogen-hunter\",\"messages\":[{{\"role\":\"user\",\"content\":\"{prompt}\"}}]}}";
            var content = new StringContent(json, Encoding.UTF8, "application/json");
            var response = await _httpClient.PostAsync($"{_endpoint}/v1/chat/completions", content);
            response.EnsureSuccessStatusCode();
            return await response.Content.ReadAsStringAsync();
        }
    }
}
